File: java/src/pursuer/pxprpc/PxpCallable.java
package pursuer.pxprpc;

import java.io.IOException;

public interface PxpCallable {
	void readParameter(PxpRequest req) throws IOException ;
	void call(PxpRequest req,AsyncReturn<Object> asyncRet) throws IOException;
	void writeResult(PxpRequest req) throws IOException;
}

File: csharp/dotnet/pxprpc/PxpCallable.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace pxprpc
{
	public interface PxpCallable
	{
		void readParameter(PxpRequest req);
        void call(PxpRequest req, Action<Object> asyncRet);
		void writeResult(PxpRequest req);
	}
}